### ISSUE TITLE
xorrisofs argument order prevents efi boot

### ISSUE THREAD

**Issue by sobjerke** [livecd-tools/livecd-tools#190]
In "imgcreate/live.py", the order of options to xorrisofs produces an image that is only bootable by bios. Booting via efi fails.

Additionally, the image will not be bootable at all if efiboot.img for some reason is missing. This file should only be needed for efi, but all El-Torito options are skipped if efiboot.img is missing.

The following changes producues images that can boot via both bios and efi:

```
diff --git a/imgcreate/live.py b/imgcreate/live.py
index ebe52ca..5d20aca 100644
--- a/imgcreate/live.py
+++ b/imgcreate/live.py
@@ -384,14 +384,13 @@ class x86LiveImageCreator(LiveImageCreatorBase):
         self._efiarch = None

     def _get_xorrisofs_options(self, isodir):
-        options = []
+        options = ["-eltorito-boot", "isolinux/isolinux.bin",
+                   "-no-emul-boot", "-boot-info-table", "-boot-load-size", "4",
+                   "-eltorito-catalog", "isolinux/boot.cat",
+                   "-isohybrid-mbr", "/usr/share/syslinux/isohdpfx.bin"]
         if os.path.exists(os.path.join(isodir, "isolinux/efiboot.img")):
             options += ["-eltorito-alt-boot", "-e", "isolinux/efiboot.img",
-                        "-no-emul-boot", "-isohybrid-gpt-basdat",
-                        "-isohybrid-mbr", "/usr/share/syslinux/isohdpfx.bin",
-                        "-eltorito-boot", "isolinux/isolinux.bin",
-                        "-boot-load-size", "4", "-boot-info-table",
-                        "-eltorito-catalog", "isolinux/boot.cat"]
+                        "-no-emul-boot", "-isohybrid-gpt-basdat"]
         if os.path.exists(os.path.join(isodir, "isolinux/macboot.img")):
             options += ["-eltorito-alt-boot", "-e", "isolinux/macboot.img",
                         "-no-emul-boot", "-isohybrid-gpt-hfsplus"]
```

I can create a PR for this, but it depens on #189

**Comment by mikhailnov** [comment]
+1 for this patch, thanks!

**Comment by sobjerke** [comment]
Resolved via PR #192
